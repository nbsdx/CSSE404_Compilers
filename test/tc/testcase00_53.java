// Invalid assignment of variable x

class Foo 
{ 
	public static void main(String[] args) 
	{ 
	}
}

class Bar 
{ 
	int x;
	
	public Bar self(int a) 
	{ 
		return this; 
	}
}

class Baz extends Bar 
{ 
	public Bar self(int a) 
	{ 
		x = true;
		return this; 
	}
}
