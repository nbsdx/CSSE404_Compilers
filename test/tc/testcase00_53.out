Cannot assign type boolean to variable x of type int
